### PR TITLE
Add new registration flow test to TSS side

### PR DESCRIPTION
**Pull request by HCastano**
While working on the signing flow I realized that there wasn't a test checking the new
registration flow on the TSS side (which is a pre-requisite to doing signing).

I've added a mostly happy-path test for registration. I don't really want to get bogged
down in checking all the failure scenarios until after the signing flow is done.

### CODE REVIEW

File: crates/threshold-signature-server/src/user/tests.rs [hunk 1]
```diff
@@ -828,6 +829,32 @@ pub async fn put_register_request_on_chain(
...
+        PairSigner::<EntropyConfig, sp_core::sr25519::Pair>::new(signature_request_account.pair());
+
+    let registering_tx =
+        entropy::tx().registry().register_on_chain(program_modification_account, program_instance);
+
+    let events =
+        submit_transaction(api, rpc, &signature_request_account, &registering_tx, None).await?;
+
+    // Since we're only submitting one request above, looking for the first event as opposed to
+    // say, all events, should be fine.
```
Comment (ameba23) [inline crates/threshold-signature-server/src/user/tests.rs:851]: This is nice.  But in our actual production client code we will still need a loop to filter out the events of other users, right?

**Review by ameba23 — APPROVED** [review]
:100: nice clear comments i gotta say

**Review by JesseAbram — APPROVED** [review]

File: crates/threshold-signature-server/src/user/tests.rs (hunk 1 above)
Comment (JesseAbram) [reply crates/threshold-signature-server/src/user/tests.rs:851]: yes that is correct

File: pallets/registry/src/lib.rs [hunk 2]
```diff
@@ -143,15 +143,6 @@ pub mod pallet {
                         version_number: T::KeyVersionNumber::get(),
                     },
                 );
-
-                RegisteredOnChain::<T>::insert(
```
[the remark this answers is not in the document]
Comment (HCastano) [reply pallets/registry/src/lib.rs:147]: For the moment no. I wrote a little thing out in the commit message removing this why it doesn't make sense to have them. tl;dr registration requires a jumpstarted network as a pre-req and that's a bit of a tricky setup using the Genesis config code.

It's something we might want to look at in the future if we find that manually setting everything up with extrinsics becomes cumbersome

**Pull request merged**